app/scripts/controllers/metametrics.ts

```diff
@@ -470,7 +498,22 @@ export default class MetaMetricsController {
     const fragment = fragments[id];
 
     if (!fragment) {
-      throw new Error(`Event fragment with id ${id} does not exist.`);
+      if (!id.includes('transaction-submitted-')) {
+        throw new Error(`Event fragment with id ${id} does not exist.`);
+      }
+
+      /**
+       * HACK: "transaction-submitted-<id>" fragment hack
+       * Creates a "transaction-submitted-<id>" fragment if it does not exist to persist
+       * accumulated event metrics. In the case it is unused, the abandoned fragment will
+       * eventually be deleted with canThrowAwayIfAbandoned set to true.
+       */
+      fragments[id] = {
+        canThrowAwayIfAbandoned: true,
+        category: MetaMetricsEventCategory.Transactions,
```

Comment:
Can this and `successEvent` be read from the `payload` or are they not provided for an update?

If so, could we provide them to avoid the coupling?